### PR TITLE
VP-6717: Add vaStoreSelector and use it in store-details blade

### PR DESCRIPTION
**Pull request by eMazeika**
https://virtocommerce.atlassian.net/browse/VP-6717

### CODE REVIEW

**Review by akak1977 — DISMISSED** [review]

File: src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.tpl.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="form-input">
+    <ui-select ng-if="context.multiple" multiple ng-model="context.modelValue" ng-class="{disabled: stores.length === 0}">
+        <ui-select-match placeholder="{{placeholder || 'stores.directives.store-selector.placeholder' | translate}}">{{stores.length > 0 ? $item.name : 'stores.directives.store-selector.loading' | translate}}</ui-select-match>
+        <ui-select-choices repeat="x.id as x in stores | filter: { name: $select.search }" refresh="fetchStores($select)" refresh-delay="300" when-scrolled="fetchNextStores($select)">
+            <span ng-bind-html="x.name | highlight: $select.search" ng-show='stores.length > 0'></span>
```
Comment (Vectorfield4) [inline src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.tpl.html:5]: Can we add a property for this condition `stores.length > 0` and replace it in the template and [here](https://github.com/VirtoCommerce/vc-module-store/blob/00cf93b4338e6aef5082f8559c843ca308d97009/src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.js#L38)?

File: src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                                totalCount = Math.max(totalCount, data.totalCount);
+                            }).$promise;
+                    }
+
+                    return $q.resolve();
+                };
+
+                function joinStores(newItems) {
+                    newItems = _.reject(newItems, x => _.any($scope.stores, y => y.id === x.id) || _.indexOf($scope.storesToHide, x.id) > -1);
+                    $scope.stores = $scope.stores.concat(newItems);
```
Comment (Vectorfield4) [inline src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.js:83]: Maybe we should push items through iteration without rewriting variable?

File: src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.tpl.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        </ui-select-choices>
+    </ui-select>
+    <ui-select ng-if="!context.multiple" ng-model="context.modelValue" ng-required="context.required" ng-class="{disabled: stores.length === 0}">
+        <ui-select-match placeholder="{{placeholder || 'stores.directives.store-selector.placeholder' | translate}}">{{stores.length > 0 ? $select.selected.name : 'stores.directives.store-selector.loading' | translate}}</ui-select-match>
+        <ui-select-choices repeat="x.id as x in stores | filter: { name: $select.search }" refresh="fetchStores($select)" refresh-delay="300" when-scrolled="fetchNextStores($select)">
+            <span ng-bind-html="x.name | highlight: $select.search" ng-show='stores.length > 0'></span>
+        </ui-select-choices>
+    </ui-select>
+</div>
```
Comment (Vectorfield4) [inline src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.tpl.html:14]: Add newline, please 🙏

**Review by Vectorfield4 — APPROVED** [review]
Nice work!

File: src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.js (hunk 2 above)
Comment (eMazeika) [reply src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.js:83]: interesting, but iterating could heart the performance

File: src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.tpl.html (hunk 1 above)
Comment (eMazeika) [reply src/VirtoCommerce.StoreModule.Web/Scripts/directives/vaStoreSelector.tpl.html]: sure

**Review by akak1977 — APPROVED** [review]

**Review by OlegoO — APPROVED** [review]

**Pull request merged**